### PR TITLE
[release-1.23] releng: Update images, dependencies and version to Go 1.19.6

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind feature
/area dependency

#### What this PR does / why we need it:

- releng: Update images, dependencies and version to Go 1.19.6

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/release/issues/2909

/assign @xmudrii @saschagrunert @dims @liggitt
cc @kubernetes/release-managers 

#### Does this PR introduce a user-facing change?
```release-note
Kubernetes is now built with Go 1.19.6
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

cc @kubernetes/release-engineering

### CODE REVIEW

File: build/dependencies.yaml [hunk 1]
```diff
@@ -107,9 +107,8 @@ dependencies:
     version: 1.19
     refPaths:
     - path: build/build-image/cross/VERSION
-    # TODO(liggitt): Uncomment once images are updated to go1.19
-    #- path: hack/lib/golang.sh
-    #  match: minimum_go_version=go([0-9]+\.[0-9]+)
```
Comment (liggitt) [inline build/dependencies.yaml:112]: revert this change, per https://github.com/kubernetes/enhancements/tree/master/keps/sig-release/3744-stay-on-supported-go-versions#proposal we'll continue to run unit and integration tests on the original go version for 1.23, which requires leaving the minimum go version in golang.sh as 1.17.x

can drop the TODO or change the comment to explain why this is 1.17

File: hack/lib/golang.sh [hunk 2]
```diff
@@ -481,7 +481,7 @@ EOF
   IFS=" " read -ra go_version <<< "$(GOFLAGS='' go version)"
   local minimum_go_version
   # TODO(liggitt): Need to switch this to 1.19 once we update images to newer go version
-  minimum_go_version=go1.17.0
+  minimum_go_version=go1.19
```
Comment (liggitt) [inline hack/lib/golang.sh:484]: revert this change, per https://github.com/kubernetes/enhancements/tree/master/keps/sig-release/3744-stay-on-supported-go-versions#proposal we'll continue to run unit and integration tests on the original go version for 1.23, which requires leaving the minimum go version in golang.sh as 1.17.x

can drop the TODO or change the comment to explain why this is 1.17

File: build/dependencies.yaml (hunk 1 above)
Comment (cpanato) [reply build/dependencies.yaml:112]: done, thanks

File: hack/lib/golang.sh (hunk 2 above)
Comment (cpanato) [reply hack/lib/golang.sh]: done

File: hack/lib/golang.sh [hunk 3]
```diff
@@ -480,8 +480,9 @@ EOF
   local go_version
   IFS=" " read -ra go_version <<< "$(GOFLAGS='' go version)"
   local minimum_go_version
-  # TODO(liggitt): Need to switch this to 1.19 once we update images to newer go version
-  minimum_go_version=go1.17.0
+  # we continue to test in the original go version with was go1.18
```
Comment (liggitt) [inline hack/lib/golang.sh:483]:
```suggestion
  # we continue to test in the original go version with was go1.17
```

Comment (cpanato) [reply hack/lib/golang.sh]: bad copy/paste :(
